rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -421,6 +424,21 @@ public void registerBtcTransaction(
         }
     }
 
+    private void processSvpFundTransactionHashSigned(BtcTransaction transaction) {
+        Optional<Sha256Hash> fundTransactionHashUnsignedOpt = provider.getSvpFundTxHashUnsigned();
+        if (!fundTransactionHashUnsignedOpt.isPresent()) {
+            return;
+        }
+        Sha256Hash fundTransactionHashUnsigned = fundTransactionHashUnsignedOpt.get();
+
+        BtcTransaction transactionCopy = new BtcTransaction(networkParameters, transaction.bitcoinSerialize()); // this is needed to not remove signatures from the actual tx
+        BitcoinUtils.removeSignaturesFromTransactionWithP2shMultiSigInputs(transactionCopy);
+        if (transactionCopy.getHash().equals(fundTransactionHashUnsigned)) {
+            provider.setSvpFundTxHashSigned(transaction.getHash());
+            provider.setSvpFundTxHashUnsigned(null);
```

Comment:
Should we also call the `save` method after this? The way I understand this is since the value only live in memory, once the rsk transaction execution ends we will loose all the values in memory. So how do we persist this data before it ends? Maybe I am understanding this wrong :)